### PR TITLE
325 alter ranking of results penalising results found at the same rank

### PR DESCRIPTION
**Pull request by yaseminbridges**
Altered the ranking of the post-processed results so equal ranks are now penalised.

Before:

1,1,3

Now

2,2,3

### CODE REVIEW

File: src/pheval/post_processing/post_processing.py [hunk 1]
```diff
@@ -227,68 +227,7 @@ def sort_pheval_results(self) -> [PhEvalResult]:
         )
 
 
-class ScoreRanker:
```
Comment (souzadevinicius) [inline src/pheval/post_processing/post_processing.py:230]: Has the class been moved?

File: tests/test_post_processing.py [hunk 2]
```diff
@@ -216,40 +217,40 @@ def test_sort_pheval_results_pvalue(self):
...
-        self.assertEqual(self.score_ranker_descending.rank_scores(0.7342), 1)
-        self.assertEqual(self.score_ranker_descending.rank_scores(0.3452), 2)
-        self.assertEqual(self.score_ranker_descending.rank_scores(0.3452), 2)
-
-    def test_rank_scores_count_increase(self):
-        self.assertEqual(self.score_ranker_descending.rank_scores(0.7342), 1)
-        self.assertEqual(self.score_ranker_descending.rank_scores(0.3452), 2)
-        self.assertEqual(self.score_ranker_descending.rank_scores(0.3452), 2)
-        self.assertEqual(self.score_ranker_descending.rank_scores(0.1234), 4)
+# class TestScoreRanker(unittest.TestCase):
```
Comment (souzadevinicius) [inline tests/test_post_processing.py:220]: Will this test return in the future?

**Comment by souzadevinicius** [comment]
> Altered the ranking of the post-processed results so equal ranks are now penalised.
> 
> Before:
> 
> 1,1,3
> 
> Now
> 
> 2,2,3

Why equal ranks are penalised?

**Review by souzadevinicius — APPROVED** [review]

**Comment by yaseminbridges** [comment]
> > Altered the ranking of the post-processed results so equal ranks are now penalised.
> > Before:
> > 1,1,3
> > Now
> > 2,2,3
> 
> Why equal ranks are penalised?

Damian and Jules agreed that this is the better way to rank results. In theory with the current ranking if a tool outputs all genes with the same score and they were all ranked 1 the results would be misleading. If it is penalised we can account for these types of instances.

File: src/pheval/post_processing/post_processing.py (hunk 1 above)
Comment (yaseminbridges) [reply src/pheval/post_processing/post_processing.py:230]: It has been removed.

File: tests/test_post_processing.py (hunk 2 above)
Comment (yaseminbridges) [reply tests/test_post_processing.py]: No, as the class has been removed.

**Pull request merged**